### ISSUE TITLE
TabPy Docker image not running in Kubernetes

### ISSUE THREAD

**Issue by andrearota** [tableau/TabPy#497]
**Environment information:**
 - Kubernetes
 - Python version: 3.8 (used python:3.8 base image)
 - TabPy: 2.3.1

**Describe the issue**
I am trying to run the Docker image on Kubernetes. When running the Docker image with docker run, it works flawlessly. 
```
docker run --rm -it da326489f806
2021-02-23,11:43:43 [INFO] (app.py:app:242): Parsing config file /usr/local/lib/python3.8/site-packages/tabpy/tabpy_server/app/../common/default.conf
2021-02-23,11:43:43 [INFO] (app.py:app:431): Loading state from state file /usr/local/lib/python3.8/site-packages/tabpy/tabpy_server/state.ini
2021-02-23,11:43:43 [INFO] (app.py:app:328): Password file is not specified: Authentication is not enabled
2021-02-23,11:43:43 [INFO] (app.py:app:343): Call context logging is disabled
2021-02-23,11:43:43 [INFO] (app.py:app:124): Initializing TabPy...
2021-02-23,11:43:43 [INFO] (callbacks.py:callbacks:43): Initializing TabPy Server...
2021-02-23,11:43:43 [INFO] (app.py:app:128): Done initializing TabPy.
2021-02-23,11:43:43 [INFO] (app.py:app:82): Setting max request size to 104857600 bytes
2021-02-23,11:43:43 [INFO] (callbacks.py:callbacks:64): Initializing models...
2021-02-23,11:43:43 [INFO] (app.py:app:105): Web service listening on port 9004
^C2021-02-23,11:44:05 [CRITICAL] (app.py:app:116): Exiting on signal 2...
2021-02-23,11:44:05 [INFO] (app.py:app:122): Shutting down TabPy...
2021-02-23,11:44:05 [INFO] (app.py:app:122): Shutting down TabPy...
```

When I run it on Kubernetes, I get the following error:

```
kubectl logs tabpy-6dbbdf5df-pb8tl
2021-02-23,12:01:02 [INFO] (app.py:app:242): Parsing config file /usr/local/lib/python3.8/site-packages/tabpy/tabpy_server/app/../common/default.conf
2021-02-23,12:01:02 [INFO] (app.py:app:431): Loading state from state file /usr/local/lib/python3.8/site-packages/tabpy/tabpy_server/state.ini
2021-02-23,12:01:02 [INFO] (app.py:app:328): Password file is not specified: Authentication is not enabled
2021-02-23,12:01:02 [INFO] (app.py:app:343): Call context logging is disabled
2021-02-23,12:01:02 [INFO] (app.py:app:124): Initializing TabPy...
2021-02-23,12:01:02 [INFO] (callbacks.py:callbacks:43): Initializing TabPy Server...
2021-02-23,12:01:02 [INFO] (app.py:app:128): Done initializing TabPy.
2021-02-23,12:01:02 [INFO] (app.py:app:82): Setting max request size to 104857600 bytes
2021-02-23,12:01:02 [INFO] (callbacks.py:callbacks:64): Initializing models...
Traceback (most recent call last):
  File "/usr/local/bin/tabpy", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/tabpy/tabpy.py", line 44, in main
    app.run()
  File "/usr/local/lib/python3.8/site-packages/tabpy/tabpy_server/app/app.py", line 98, in run
    application.listen(
  File "/usr/local/lib/python3.8/site-packages/tornado/web.py", line 2109, in listen
    server.listen(port, address)
  File "/usr/local/lib/python3.8/site-packages/tornado/tcpserver.py", line 151, in listen
    sockets = bind_sockets(port, address=address)
  File "/usr/local/lib/python3.8/site-packages/tornado/netutil.py", line 104, in bind_sockets
    socket.getaddrinfo(address, port, family, socket.SOCK_STREAM, 0, flags),
  File "/usr/local/lib/python3.8/socket.py", line 918, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -8] Servname not supported for ai_socktype
```

**To Reproduce**

Deploy TabPy image on your container registry, then apply the following manifest.

```yaml
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: null
  labels:
    run: po
  name: po
spec:
  containers:
  - args:
    - tabpy
    image: <your-registry>/tabpy:<version>
    name: po
    ports:
    - containerPort: 9004
    resources: {}
  dnsPolicy: ClusterFirst
  restartPolicy: Always
status: {}
```

**Expected behavior**
Container running on Kubernetes

**Comment by andrearota** [comment]
The solution is to set an environment variable for the TabPy container, `TABPY_PORT="9004"`.

```yaml
[...]
      containers:
        - name: tabpy
          image: <your-registry>/tabpy:<version>
          env:
            - name: TABPY_PORT
              value: "9004"
```

If not set, the container will read the port variable as the URI of the endpoint (probably as injected by Kubernetes, being both the service and the pod named tabpy).

```2021-02-23,15:14:53 [DEBUG] (app.py:app:195): Parameter port set to "tcp://172.16.19.10:80" from config file or environment variable```

TL;DR: be aware of calling your pod/deployment/service "tabpy". In case you are forced to do so, remember to override the value of the environment variable TABPY_PORT to "9004".
